### PR TITLE
See comments

### PR DESCRIPTION
**Pull request by Or-Tal**
removed demucs hifi from master. re-starting this implementation.
I've also added @Package to the config files so they would work with the shared environment.
also changed conf/config.yaml to conf/main_config.yaml as it is needed to support use of hubert model (which load it's own config.yaml from the internet in the background)
---
I've fixed demucs_hifi and it appears to be working a lot better.

### CODE REVIEW

**Review by mosem — COMMENTED** [review]
Hey,
Look at my comments.
Besides some comments about naming and redundant code, I'm suggesting a redesign of both demucsHifi and of the demucsHifIBS classes, that will fit the existing code.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -30,11 +30,10 @@ jobs:
         ./make_debug.sh
         python train.py num_workers=1 device=cpu epochs=1 experiment=caunet_1 experiment.scale_factor=2 experiment.caunet.hidden=8 experiment.caunet.depth=3 experiment.caunet.dense_block_depth=2
 
-
-    - name: Demucs HiFi - dummy full pipeline
+    - name: DemucsHifi, no MRF with Disc - dummy full pipeline
       run: |
         ./make_debug.sh
-        python train.py num_workers=2 device=cpu epochs=1 experiment=demucs_hifi experiment.scale_factor=2 eval_every=1
+        python train.py num_workers=1 device=cpu epochs=1 experiment.experiment_name=demucs_hifi_no_mrf_with_disc_1 experiment=demucs_hifi experiment.demucs_hifi.num_mrfs=0 experiment.demucs_hifi_bs.l1_factor=100 experiment.demucs_hifi_bs.gen_factor=2 experiment.demucs_hifi.depth=2
```
Comment (mosem) [inline .github/workflows/tests.yml:36]: make sure that the test runs in minimal time.

**Review by mosem — COMMENTED** [review]
.

File: denoiser/batch_solvers/demucs_hifi_bs.py [hunk 2]
```diff
@@ -1,157 +1,198 @@
...
 
-        if not cross_valid:
-            optim_d.zero_grad()
+def discriminator_loss(disc_real_outputs, disc_generated_outputs):
+    loss = 0
+    r_losses = []
+    g_losses = []
+    for dr, dg in zip(disc_real_outputs, disc_generated_outputs):
+        r_loss = torch.mean((1-dr)**2)
+        g_loss = torch.mean(dg**2)
```
[the remark this answers is not in the document]
Comment (Or-Tal) [reply denoiser/batch_solvers/demucs_hifi_bs.py:39]: changed to disc_real and disc_generated as they imply

File: denoiser/batch_solvers/demucs_hifi_bs.py [hunk 3]
```diff
@@ -1,157 +1,198 @@
...
+            estimate = predicted
             x_ft = 0
 
-        # take discriminator step
-        loss_disc_all = self._disc_step(mpd, msd, y, y_g_hat, cross_valid, optim_d)
+        audio_loss = F.l1_loss(clean, estimate) * self.l1_factor
+
+        y_df_hat_r, y_df_hat_g, fmap_f_r, fmap_f_g = self._models[MPD](clean, estimate.detach())
+        y_ds_hat_r, y_ds_hat_g, fmap_s_r, fmap_s_g = self._models[MSD](clean, estimate.detach())
+        loss_fm_f = feature_loss(fmap_f_r, fmap_f_g) * self.gen_factor
```
[the remark this answers is not in the document]
Comment (Or-Tal) [reply denoiser/batch_solvers/demucs_hifi_bs.py:137]: that's sufficiently informative, taken out of source.

**Pull request closed without merging**